packages/core/src/services/time-series.ts

```diff
@@ -0,0 +1,132 @@
+import { format } from "date-fns";
+import * as locales from "date-fns/locale";
+import { TimeScaleOptions } from "../interfaces/axis-scales";
+import { Tools } from "../tools";
+
+// D3 Imports
+import { min } from "d3-array";
+
+export const TIME_INTERVALS = [
+	["15seconds", 15 * 1000],
+	["minute", 60 * 1000],
+	["30minutes", 30 * 60 * 1000],
+	["hourly", 60 * 60 * 1000],
+	["daily", 24 * 60 * 60 * 1000],
+	["monthly", 30 * 24 * 60 * 60 * 1000],
+	["quarterly", 3 * 30 * 24 * 60 * 60 * 1000],
+	["yearly", 12 * 30 * 24 * 60 * 60 * 1000]
+];
+
+const codes = Object.values(locales).map(locale => locale["code"]);
+
+// Return true if the tick is a primary tick, false otherwise
+export function isTickPrimary(tick: number, i: number, interval: string, showDayName: boolean): boolean {
+	const isFirstTick = i === 0;
+	const hasANewWeekStarted = Number(format((new Date(tick)), "c")) === 2;
+	const isFirstQuarter = Number(format((new Date(tick)), "q")) === 1;
+
+	switch (interval) {
+		case "15seconds": return (isFirstTick || isDayOfMonthChanged(tick) || isMonthChanged(tick) || isYearChanged(tick));
+		case "minute": return (isFirstTick || isDayOfMonthChanged(tick) || isMonthChanged(tick) || isYearChanged(tick));
+		case "30minutes": return (isFirstTick || isDayOfMonthChanged(tick) || isMonthChanged(tick) || isYearChanged(tick));
+		case "hourly": return (isFirstTick || isDayOfMonthChanged(tick) || isMonthChanged(tick) || isYearChanged(tick));
+		case "daily":
+			if (!showDayName) { // daily
+				return (isFirstTick || isMonthChanged(tick) || isYearChanged(tick));
+			} else { // weekly
+				return (isFirstTick || hasANewWeekStarted || isYearChanged(tick));
+			}
+		case "monthly": return (isFirstTick || isYearChanged(tick));
+		case "quarterly": return (isFirstTick || isFirstQuarter);
+		case "yearly": return false;
+		default: throw new Error(`${interval} is not a valid time interval.`);
+	}
+}
+
+// The accepted formats of localeCode are ll and ll-CC
+function getLocale(localeCode: string): Locale {
+	// locales is an object whose keys format is ll or llCC
+	// each locale is an object whose code value format is ll or ll-CC
+	const localeCodeWithoutDash = localeCode.replace(/-/g, "");
+	const foundLocale = locales[localeCodeWithoutDash];
+	if (!foundLocale) {
+		throw new Error(`Locale with code ${localeCode} not found. Avaible codes are: ${codes}.`);
+	}
+	return foundLocale;
+}
+
+// Return the formatted current tick
+export function formatTick(tick: number, i: number, interval: string, timeScaleOptions: TimeScaleOptions): string {
+	const showDayName = timeScaleOptions.showDayName;
```

Comment:
Are we going to support per-axis timescale options or the global `timeScale` field?

Is there a better/cleaner way? Are we breaking anything?